### DISCUSSION TITLE
Expected element type <AssumeRoleResponse> but have <ErrorResponse> | The request signature we calculated does not match the signature you provided. Check your key and signing method.

### DISCUSSION THREAD

**Discussion by cubuzz** [minio/minio#16833] [Q&A] [answered]
Heya :)

Recently re-installed minio on my server setup and I can no longer operate it as per usual.

I've set up the proxies https://console.cdn to point to the minio console, as well as https://cdn to point to the minio api. Both are reachable from outside. Minio, under the hood, listens to the local loopback, which then in turn gets proxied by nginx.

My "default" file as suggested per docs is the following:

```
MINIO_OPTS="--address 127.0.0.1:9000 --console-address=127.0.0.1:9001"

MINIO_ROOT_USER=admin
MINIO_ROOT_PASSWORD=password

MINIO_VOLUMES="/srv/cdn"

MINIO_SERVER_URL="https://cdn"
MINIO_BROWSER_REDIRECT_URL="https://console.cdn"
```

I have seen https://github.com/minio/minio/discussions/14077 , but I can't seem to figure out if I have the same issue or not.
I *can* log in if I remove the server URL from the defaults, however the CLI is also complaining about: `The request signature we calculated does not match the signature you provided. Check your key and signing method.` So neither way is currently fully working.

Any ideas would be appreciated :)

**Comment by vadmeste** [comment, marked as answer]
@cubuzz if you are using nginx, they are some configuration that need to be set, https://min.io/docs/minio/linux/integrations/setup-nginx-proxy-with-minio.html

nginx modifies 'Host' header sent by client and this will issue signature mismatch error.

**Reply by cubuzz** [reply]
Thank you very much! I somehow did not catch that. After applying the changes and ironing out some minor problems, it's working perfectly now.

**Reply by wherelse** [reply]
> @cubuzz if you are using nginx, they are some configuration that need to be set, https://min.io/docs/minio/linux/integrations/setup-nginx-proxy-with-minio.html
> 
> nginx modifies 'Host' header sent by client and this will issue signature mismatch error.

Hello, Do you have experience in using Caddy? I tried to modify my caddy configuration by referring to nginx configuration, but it has been unsuccessful, still encounter the same error.
this is my caddy config:
```Caddyfile
xxxx.xxx.com {
	reverse_proxy {
		to localhost:9001
		header_up X-Real-IP {http.request.remote}
		header_up X-Forwarded-Host {remote}
		header_up Host {http.reverse_proxy.upstream.hostport}
	}
	reverse_proxy /datas {
		to localhost:9000
		header_up X-Real-IP {http.request.remote}
		header_up X-Forwarded-Host {remote}
		header_up Host {http.reverse_proxy.upstream.hostport}
	}
	header Cache-Control max-age=63072000
	header / {
		X-Frame-Options "DENY"
		X-Robots-Tag "none"
		-Server
	}
	encode zstd gzip
}
```
Thank you very much if you can help me.

**Comment by wherelse** [comment]
I met the same problem with you：`Expected element type <AssumeRoleResponse> but have <ErrorResponse>`,  did you solve the problem?

**Reply by cubuzz** [reply]
Yep! It was as @vadmeste said a series of misconfigurations within nginx. Check the docs mentioned, you might need to do some more tweaking with proxy_buffering.
